components/cloud_subscribe_with_loading_modal/index.tsx

```diff
@@ -0,0 +1,211 @@
+            updateProgress,
+            MIN_PROCESSING_MILLISECONDS / MAX_FAKE_PROGRESS,
+        );
+    }, []);
+
+    const handleSubscribe = async () => {
+        const start = new Date();
+        if (props.teamToKeep) {
+            await dispatch(archiveAllTeamsExcept(props.teamToKeep.id));
+        }
+
+        const productUpdated = await dispatch(subscribeCloudSubscription(
+            props.selectedProduct?.id as string,
+        ));
+
+        // the action subscribeCloudSubscription returns a true boolean when successful and an error when it fails
+        if (typeof productUpdated !== 'boolean') {
+            setError(true);
+            setProcessingState(ProcessState.FAILED);
+            return;
+        }
+
+        const end = new Date();
+        const millisecondsElapsed = end.valueOf() - start.valueOf();
+        if (millisecondsElapsed < MIN_PROCESSING_MILLISECONDS) {
+            setTimeout(
+                completeSubscribe,
+                MIN_PROCESSING_MILLISECONDS - millisecondsElapsed,
+            );
+            return;
+        }
+        completeSubscribe();
+    };
+
+    const completeSubscribe = () => {
+        clearInterval(intervalId.current);
+        setProcessingState(ProcessState.SUCCESS);
+        setProgress(100);
+    };
+
+    const updateProgress = () => {
+        if (progress >= MAX_FAKE_PROGRESS) {
+            clearInterval(intervalId.current);
+            return;
+        }
+
+        setProgress((progress) => {
+            if (progress >= MAX_FAKE_PROGRESS) {
+                clearInterval(intervalId.current);
+            }
+            return progress + 3 > MAX_FAKE_PROGRESS ? MAX_FAKE_PROGRESS : progress + 3;
+        },
+        );
+    };
+
+    const handleGoBack = () => {
+        clearInterval(intervalId.current);
+        setProgress(0);
+        setError(false);
+        setProcessingState(ProcessState.PROCESSING);
+        props.onBack();
+    };
+
+    const handleClose = () => {
+        dispatch(
+            closeModal(
+                ModalIdentifiers.CLOUD_SUBSCRIBE_WITH_LOADING_MODAL,
+            ),
+        );
+        if (typeof props.onClose === 'function') {
+            props.onClose();
+        }
+    };
+
+    const successPage = () => {
+        const formattedBtnText = (
+            <FormattedMessage
+                defaultMessage='Return to {team}'
+                id='admin.billing.subscription.returnToTeam'
+                values={{team: props.teamToKeep?.display_name}}
+            />
+        );
+        const productName = props.selectedProduct?.name;
+        const title = (
+            <FormattedMessage
+                id={'admin.billing.subscription.downgradedSuccess'}
+                defaultMessage={"You're now subscribed to {productName}"}
+                values={{productName}}
+            />
+        );
+
+        const formattedSubtitle = (
+            <FormattedMessage
+                id='success_modal.subtitle'
```

Comment:
It looks like this translation key is reused in `components/cloud_subscribe_result_modal/success.tsx` with a different message, so one of those is going to be translated incorrectly to use the other's text